actors/miner/tests/util.rs

```diff
@@ -3417,6 +3345,39 @@ pub fn make_piece_manifest(
     }
 }
 
+pub fn make_piece_specs_from_configs(
+    sector_number: u64,
+    deal_ids: &Vec<DealID>,
+    prove_cfg: &ProveCommitConfig,
+) -> Vec<(u64, ActorID, AllocationID, DealID)> {
+    static EMPTY_VEC: Vec<ActivatedDeal> = Vec::new();
+    let configured_deals = prove_cfg.activated_deals.get(&sector_number).unwrap_or(&EMPTY_VEC);
+    // The old configuration system had duplicated information between cfg and precommit inputs
+    // To ensure passed in configuration is internally consistent check that cfg deals are a subset
+    // of precommitted deal ids
```

Comment:
Can you explain why it's a subset, and why below we can safely `get(i)` of `configured_deals`? So it's a subset, but strictly has a copy of the deals referenced in the prefix of `deal_ids`. Why?

Reply:
or ... is it simply that we want to take `configured_deals` and _assign_ them a deal_id so we pluck them out in order, and then fill the rest up with defaults?

Reply:
I guess I'm struggling to see why there's additional deal_ids involved here, what are the test cases that involve a different number of deal_ids vs what's in the config?

Reply:
> or ... is it simply that we want to take configured_deals and assign them a deal_id so we pluck them out in order, and then fill the rest up with defaults?

Yes this is it.

> I guess I'm struggling to see why there's additional deal_ids involved here

Yeah you and me both 😄 
I don't know if there are any tests that configure more deal ids than deal specs in ProveCommitCfgs but I doubt they're there.  However the two configs are completely logically separate so it is technically possible and I wanted to handle just in case.

I could alternatively look into failing the test if the configs don't have the same number of deals but if I hit any failure cases I'd want to just cleanup the mess with #1545.